src/realm/object-store/shared_realm.cpp

```diff
@@ -232,14 +233,16 @@ void Realm::read_schema_from_group_if_needed()
         return;
     }
 
+    auto schema = m_schema;
     Group& group = read_group();
     auto current_version = transaction().get_version_of_current_transaction().version;
-    if (m_schema_transaction_version == current_version)
```

Comment:
This is a significant pessimization. Reading the schema is quite slow and we need to avoid doing it as much as possible.

Reply:
I think this is a mistake, this line should have not been deleted.